frontend/src/metabase/visualizations/visualizations/SmartScalar/compute.js

```diff
@@ -431,14 +439,14 @@ function getRowOfPeriodsAgo({
   // skip the latest element since that is our current value
   const searchIndexStart = nextValueRowIndex - 1;
   if (searchIndexStart < 0) {
-    return -1;
+    return undefined;
   }
 
   // only look dateUnitsAgo elements (dates) into the past,
   // since looking any further would automatically result in a date before
   // X periods ago and any prior dates would be further beyond our desired
   // comparison date
-  const lastCandidateIndex = nextValueRowIndex - 1 - (dateUnitsAgo - 1);
+  const lastCandidateIndex = searchIndexStart - (dateUnitsAgo - 1);
```

Comment:
No logic change here. Just taking the `searchIndexStart` variable from up above, since it represents this value.